extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/security/HttpSecurity.java

```diff
@@ -75,10 +76,19 @@
  * }
  * </pre>
  */
+@Experimental("Remains to be determined if this is the best possible API for users to configure HTTP security")
 public interface HttpSecurity {
 
     /**
-     * Creates {@link HttpPermission}.
+     * Registers given {@link HttpAuthenticationMechanism} in addition to all other global authentication mechanisms.
+     *
+     * @param mechanism {@link HttpAuthenticationMechanism}
+     * @return HttpSecurity
+     */
+    HttpSecurity mechanism(HttpAuthenticationMechanism mechanism);
```

Comment:
@michalvavrik 
I propose to add a few shortcuts here, similarly to how you do it with `HttpPermission` below.

`HttpSecurity basic()`, or `HttpSecurity basic(String realm)` and may be a typed form option `HttpSecurity form(FormAuthenticationMechanism)` - we can do it later, just thinking that for build in mechanisms, we should users add them with some type specific support.

You can add it now or later